app/controllers/api/v1/games/shots_controller.rb

```diff
@@ -1,11 +1,12 @@
 class Api::V1::Games::ShotsController < ApiController
-  def create
-    game = Game.find(params[:game_id])
-    player = User.find_by(api_key: request.headers['X-API-Key'])
 
-    turn_processor = TurnProcessor.new(game, params[:shot][:target], player)
+  def create
+    @player = User.find_by(api_key: request.headers['X-API-Key'])
+    raise ApiExceptions::InvalidApiKey.new('Unauthorized', 401) if @player.nil?
```

Comment:
@anon0mys Think we can extract this `@player` concept to be a helper at the controller layer? If we reference a method instead it'll make testing easier and should also DRY up the code.

Reply:
That is a great suggestion!